code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -56,23 +57,38 @@
 
 		if(trash) //Do we have somehing defined as trash for our snack item ?
 			//Note : This makes sense in some way, or at least this works, just don't mess with it
+
+			//If trash is a path (like /obj/item/banana_peel), create a new object
+			//If trash is an object, use the object
+
+			//If the food item was in somebody's hands when it was eaten, put the trash item into their hands
+			//Otherwise, put the trash item in the same place where the food item was
 			if(ispath(trash, /obj/item))
-				var/obj/item/TrashItem = new trash(user)
-				user.put_in_hands(TrashItem)
+				var/obj/item/TrashItem = new trash(old_loc)
```

Comment:
Nope, it'll put the trash in the eater's hands as intended (unless the food was lying on the ground, in which case the trash will be there)